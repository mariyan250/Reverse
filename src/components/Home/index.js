// React and Style
import React, { useContext } from 'react';
import './index.scss';

// Context
import { UserContext } from '../App';

const Home = () => {
  const { user } = useContext(UserContext);

  return (
    <main className="home">
      <h1>Welcome {user.email}!</h1>
    </main>
  );
};

export default Home;