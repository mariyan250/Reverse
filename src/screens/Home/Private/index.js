// React and Style
import React from 'react';
import './index.scss';

const HomePrivate = () => {
  return (
    <main className='home-private'>
      <h1 className='text-center'>Home Logged!</h1>
    </main>
  );
};

export default HomePrivate;
