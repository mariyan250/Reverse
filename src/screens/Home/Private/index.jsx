// React and style
import React, { useState, useEffect } from 'react';
import './index.scss';

// Routes
import { searchRoutes } from 'shared/constants/routes';

// Redux
import { connect } from 'react-redux';
import { getPromoPosts } from 'store/actions/post';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import Header from 'shared/components/Header';
import CategoryCard from './components/CategoryCard';
import ItemList from 'screens/SearchProduct/components/ItemList';

const HomePrivate = ({ getPromoPosts, promoPosts }) => {
  const [searchQuery, setSearchQuery] = useState('');
  const { t } = useTranslation();

  useEffect(() => {
    getPromoPosts();
  }, [getPromoPosts]);

  const categoryCards = [
    {
      icon: '/img/home.svg',
      title: 'home_screen.private.categories.titles.home',
      url: searchRoutes.home,
      id: 1,
    },
    {
      icon: '/img/technology.svg',
      title: 'home_screen.private.categories.titles.technology',
      url: searchRoutes.technology,
      id: 2,
    },
    {
      icon: '/img/automobile.svg',
      title: 'home_screen.private.categories.titles.automobile',
      url: searchRoutes.automobile,
      id: 3,
    },
    {
      icon: '/img/sport.svg',
      title: 'home_screen.private.categories.titles.sport',
      url: searchRoutes.sport,
      id: 4,
    },
    {
      icon: '/img/tshirt.svg',
      title: 'home_screen.private.categories.titles.cloths',
      url: searchRoutes.cloths,
      id: 5,
    },
    {
      icon: '/img/pets.svg',
      title: 'home_screen.private.categories.titles.pets',
      url: searchRoutes.pets,
      id: 6,
    },
  ];

  return (
    <>
      <Header />
      <main className="home-private px-md-5">
        <form className="mx-auto my-5 px-4" action="/search">
          <article className="d-flex">
            <input
              type="text"
              name="query"
              placeholder={t('home_screen.private.input.placeholder')}
              onChange={(e) => setSearchQuery(e.target.value)}
              value={searchQuery}
              required
            />
            <article>
              <button
                className={`h-100 border-0 ${
                  searchQuery ? 'app-bg-primary' : 'app-bg-secondary'
                } d-flex justify-content-center align-items-center`}
              >
                <i className="fas fa-search app-text-white" />
              </button>
            </article>
          </article>
        </form>

        <section className="home-one px-4">
          <h2 className="mb-4">{t('home_screen.private.heading')}</h2>
          <article className="card-container">
            {categoryCards.map((card) => (
              <CategoryCard key={card.id} {...card} />
            ))}
          </article>
        </section>

        <section className="home-two mt-4">
          <ItemList items={promoPosts} title="home_screen.private.second_heading" />
        </section>
      </main>
    </>
  );
};

const mapStateToPros = (state) => ({
  promoPosts: state.posts.items,
});

export default connect(mapStateToPros, { getPromoPosts })(HomePrivate);
