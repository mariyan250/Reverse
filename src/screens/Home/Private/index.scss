@import '../../../index';

.home-private {
  .search {
    input {
      font-size: 16px;

      @include media-breakpoint-md-up {
        padding: 0.5rem !important;
      }
    }

    button {
      animation: appear 1 0.8s;
      height: 100%;
      width: 3.5rem;
      border: none;
      background: $background-red;
      border-radius: 0.2rem;
      margin-left: -2px;

      @include media-breakpoint-md-up {
        padding: 0.8rem 1rem;
        top: -0.5px;
      }
    }

    i {
      color: $white;
      font-size: 20px;
    }
  }
}
