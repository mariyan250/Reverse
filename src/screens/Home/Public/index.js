// React and style
import React, { useState } from 'react';
import './index.scss';

// Components
import Card from './components/Card';
import Arrow from './components/Arrow';
import Hamburger from 'shared/components/Hamburger';
import AuthModal from './components/AuthModal';

// Translation
import { useTranslation } from 'react-i18next';

const Home = () => {
  const [modalOpened, setModalOpened] = useState(false);
  const { t } = useTranslation();

  const cards = [
    { iconName: 'fas fa-check', title: t('home_screen.public.cards.one') },
    { iconName: 'fas fa-truck', title: t('home_screen.public.cards.two') },
    { iconName: 'fas fa-user', title: t('home_screen.public.cards.three') },
  ];

  return (
    <main className="home">
      {modalOpened && <AuthModal className="auth-m position-absolute" />}

      <section className="home-intro">
        <section className="black-layer d-flex flex-column justify-content-center align-items-center vh-100 app-bg-black-transparent">
          <h1 className="text-center font-weight-bold app-text-white">
            {t('home_screen.public.heading')}
            <span className="d-inline-block px-2 my-2 ml-3 app-bg-primary">
              {t('home_screen.public.app_name')}
            </span>
          </h1>
          <h3 className="text-center mt-1 mb-5 app-text-white">
            {t('home_screen.public.second_heading')}
          </h3>

          <Arrow className="pulsing-arrow position-absolute" />

          <Hamburger
            className={`burger-menu position-absolute ${modalOpened ? 'toggled' : ''}`}
            onClick={() => setModalOpened(!modalOpened)}
          />
        </section>
      </section>

      <section className="one pb-4">
        <hr className="my-5 my-md-5 mx-auto app-bg-primary" />
        <article className="d-md-flex flex-column">
          <h2 className="text-center">{t('home_screen.public.section_one.heading')}</h2>
          <article className="d-flex flex-column flex-md-row justify-content-between">
            <p className="p-0 mb-4">{t('home_screen.public.section_one.content')}</p>
            <img
              src="/img/cloths.jpg"
              alt={t('home_screen.public.img.alt')}
              className="img-fluid align-self-center"
            />
          </article>
        </article>
      </section>

      <section className="two py-0 app-bg-gray">
        <article className="py-3 py-md-5 d-flex flex-column flex-md-row align-items-center align-items-md-start justify-content-around">
          {cards.map((card) => (
            <Card {...card} />
          ))}
        </article>
      </section>
    </main>
  );
};

export default Home;
