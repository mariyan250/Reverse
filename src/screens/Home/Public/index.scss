@import '../../../index';

.home {
  .home-intro {
    background: url('/img/home.jpg');
    background-position: 10%;
    background-size: cover;

    .black-layer {
      background: $background-black-transparent;

      h2,
      h3 {
        color: $text-white;
      }

      h2 {
        font-size: 48px;

        span {
          background: $background-red;
        }
      }

      h3 {
        font-size: 28px;
      }
    }

    .pulsing-arrow {
      bottom: 2rem;

      @include media-breakpoint-md-up {
        bottom: 3rem;
      }
    }
  }

  .two {
    article {
      @include media-breakpoint-md-up {
        height: 100% !important;
      }
    }
  }
}
