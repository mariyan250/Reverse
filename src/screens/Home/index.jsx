// React and style
import React, { useEffect } from 'react';
import './index.scss';

// Routes
import { searchRoutes } from 'shared/constants/routes';

// Redux
import { connect } from 'react-redux';
import { getPromoPosts } from 'store/actions/post';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import CategoryCard from './components/CategoryCard';
import ItemList from 'shared/components/Product/CardList';
import Search from 'shared/components/Search';

function Home({ getPromoPosts, promoPosts }) {
  const { t } = useTranslation();

  useEffect(() => {
    getPromoPosts();
  }, [getPromoPosts]);

  const categoryCards = [
    {
      icon: '/img/home.svg',
      title: 'home_screen.categories.titles.home',
      url: searchRoutes.home,
    },
    {
      icon: '/img/technology.svg',
      title: 'home_screen.categories.titles.technology',
      url: searchRoutes.technology,
    },
    {
      icon: '/img/automobile.svg',
      title: 'home_screen.categories.titles.automobile',
      url: searchRoutes.automobile,
    },
    {
      icon: '/img/sport.svg',
      title: 'home_screen.categories.titles.sport',
      url: searchRoutes.sport,
    },
    {
      icon: '/img/tshirt.svg',
      title: 'home_screen.categories.titles.cloths',
      url: searchRoutes.cloths,
    },
    {
      icon: '/img/pets.svg',
      title: 'home_screen.categories.titles.pets',
      url: searchRoutes.pets,
    },
  ];

  return (
    <main className="home mx-auto px-4 px-md-5" data-testid="home">
      <section className="my-5 d-flex">
        <Search />
      </section>

      <section className="categories">
        <h3 className="mb-4">{t('home_screen.heading')}</h3>

        <article className="categories-container">
          {categoryCards.map((category, i) => (
            <CategoryCard url={category.url} icon={category.icon} title={category.title} key={i} />
          ))}
        </article>
      </section>

      <section className="promo-offers">
        <h3 className="mt-5 mb-4">{t('home_screen.second_heading')}</h3>

        <article className="promo-offers-container p-2 app-bg-gray-darker">
          <ItemList items={promoPosts} />
        </article>
      </section>
    </main>
  );
}

const mapStateToPros = (state) => ({
  promoPosts: state.posts.items,
});

export default connect(mapStateToPros, { getPromoPosts })(Home);
