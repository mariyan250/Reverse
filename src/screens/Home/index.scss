@import '../../index';

.home {
  max-width: $layout-max-width;

  .categories {
    .categories-container {
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      grid-gap: to-rem(22px);

      @include media-breakpoint-sm-up {
        grid-template-columns: repeat(3, 1fr);
      }

      @include media-breakpoint-md-up {
        grid-template-columns: repeat(4, 1fr);
      }

      @include media-breakpoint-lg-up {
        grid-template-columns: repeat(5, 1fr);
      }

      @include media-breakpoint-xl-up {
        grid-template-columns: repeat(6, 1fr);
      }
    }
  }
}
