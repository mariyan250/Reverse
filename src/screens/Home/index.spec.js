import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import Home from './index';

describe('Home', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { container } = renderWithRouter(<Home />, initialState);

      expect(container.querySelector('.home')).toMatchSnapshot();
    });
  });
});
