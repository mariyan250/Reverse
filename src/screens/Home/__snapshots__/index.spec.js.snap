// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Rendering should render the component with initial state 1`] = `null`;
